### DISCUSSION TITLE
how to make  custom feature matrices in archr object

### DISCUSSION THREAD

**Discussion by yingyuan830** [GreenleafLab/ArchR#1614] [Questions / Documentation]
I'd like to add a new peak-by-cell matrix to the archr object for the next step marker peak analysis, but not very sure how to make it work, can anyone give some hints ? Thanks! BTW, for the peak-by-cell matrix, if I have got the merged peak bed file, what is a reasonable method to count if a peak range of one cell is in the merged peak range ?

**Comment by rcorces** [comment]
change your peak set to whatever you want using `addPeakSet()` and then re-run `addPeakMatrix()`. You can only have one peak matrix at a time.

**Reply by yingyuan830** [reply]
For the peak set, can I just give the chrome name and start-end information, but ignore other parts ? I am not very sure how this will influence the later analysis..

**Reply by yingyuan830** [reply]
There are some errors when I run addPeakMatrix() after I use addPeakSet() on my own peakset, can you please help me ? Thanks!

Warning message in mclapply(..., mc.cores = threads, mc.preschedule = preschedule):
“8 function calls resulted in an error”
Error in (function (..., threads = 1, preschedule = FALSE) : 
Error Found Iteration 1 : 
	[1] "Error in .logError(e, fn = \".addFeatureMatrix\", info = prefix, errorList = errorList,  : \n  Exiting See Error Above\n"
	<simpleError in .logError(e, fn = ".addFeatureMatrix", info = prefix, errorList = errorList,     logFile = logFile): Exiting See Error Above>
Error Found Iteration 2 : 
	[1] "Error in .logError(e, fn = \".addFeatureMatrix\", info = prefix, errorList = errorList,  : \n  Exiting See Error Above\n"
	<simpleError in .logError(e, fn = ".addFeatureMatrix", info = prefix, errorList = errorList,     logFile = logFile): Exiting See Error Above>
Error Found Iteration 3 : 
	[1] "Error in .logError(e, fn = \".addFeatureMatrix\", info = prefix, errorList = errorList,  : \n  Exiting See Error Above\n"
	<simpleError in .logError(e, fn = ".addFeatureMatrix", info = prefix, errorList = errorList,     logFile = logFile): Exiting See Error Above>
Error Found Iteration 4 : 
	[1] "Error in .logError(e, fn = \".addFeatureMatrix\", info = prefix, errorList = errorList,  : \n  Exiting See Error Above\n"
	<simpleError in .logError(e, fn = ".addFeatureMatrix", info = prefix, errorList = errorList,     logFile = logFile): Exiting See Error Above>
Error Found Iteration 5 : 
	[1] "Error in .logError(e, fn = \".addFeatureMatrix\", info = prefix, errorList = errorList,  : \n  Exiting See Error Above\n"
	<simpleError in .logError(e, fn = ".addFeatureMatrix", info = prefix, errorList = errorList,     logFile = logFile): Exiting See Error Above>
Error Found Iteration 6 : 
	[1] "Error in .logError(e, fn = \".addFeatureMatrix\", info = prefix, errorList = errorList,  : \n  Exiting See Error Above\n"
	<simpleError in .logError(e, fn = ".addFeatureMatrix", info = prefix, errorList = errorList,     logFile = logFile): Exiting See Error Above>
Error Found Iteration 7 : 
	[1] "Error in .logError(e, fn = \".addFeatureMatrix\", info = prefix, errorList = errorList,  : \n  Exiting See Error Above\n"
	<simpleError in .logError(e, fn = ".addFeatureMatrix", info = prefix, errorList = errorList,     logFile = logFile): Exiting See Error Above>
Error Found Iteration 8 : 
	[1] "Error in .logError(e, fn = \".addFeatureMatrix\", info = prefix, errorList = errorList,  : \n  Exiting See Error Above\n"
	<simpleError in .logError(e, fn = ".addFeatureMatrix", info = prefix, errorList = errorList,     logFile = logFile): Exiting See Error Above>

Traceback:

1. addPeakMatrix(proj)
2. .batchlapply(args)
3. do.call(.safelapply, args)
4. (function (..., threads = 1, preschedule = FALSE) 
 . {
 .     if (tolower(.Platform$OS.type) == "windows") {
 .         threads <- 1
 .     }
 .     if (threads > 1) {
 .         .requirePackage("parallel", source = "cran")
 .         o <- mclapply(..., mc.cores = threads, mc.preschedule = preschedule)
 .         errorMsg <- list()
 .         for (i in seq_along(o)) {
 .             if (inherits(o[[i]], "try-error")) {
 .                 capOut <- utils::capture.output(o[[i]])
 .                 capOut <- capOut[!grepl("attr\\(\\,|try-error", 
 .                   capOut)]
 .                 capOut <- head(capOut, 10)
 .                 capOut <- unlist(lapply(capOut, function(x) substr(x, 
 .                   1, 250)))
 .                 capOut <- paste0("\t", capOut)
 .                 errorMsg[[length(errorMsg) + 1]] <- paste0(c(paste0("Error Found Iteration ", 
 .                   i, " : "), capOut), "\n")
 .             }
 .         }
 .         if (length(errorMsg) != 0) {
 .             errorMsg <- unlist(errorMsg)
 .             errorMsg <- head(errorMsg, 50)
 .             errorMsg[1] <- paste0("\n", errorMsg[1])
 .             stop(errorMsg)
 .         }
 .     }
 .     else {
 .         o <- lapply(...)
 .     }
 .     o
 . })(ceiling = 4, binarize = FALSE, verbose = TRUE, threads = 8L, 
 .     force = TRUE, logFile = "ArchRLogs/ArchR-addPeakMatrix-2f155b0a58e2-Date-2022-09-13_Time-20-11-12.log", 
 .     ArrowFiles = c(FezF2 = "/ArchR/inte4/ArrowFiles/FezF2.arrow", 
 .     Unc5b = "/ArchR/inte4/ArrowFiles/Unc5b.arrow", 
 .     PlxnD1 = "/ArchR/inte4/ArrowFiles/PlxnD1.arrow", 
 .     PV = "/ArchR/inte4/ArrowFiles/PV.arrow", 
 .     vGlut1 = "/ArchR/inte4/ArrowFiles/vGlut1.arrow", 
 .     TLE4 = "/ArchR/inte4/ArrowFiles/TLE4.arrow", 
 .     Gad2 = "/ArchR/inte4/ArrowFiles/Gad2.arrow", 
 .     Cux1 = "/ArchR/inte4/ArrowFiles/Cux1.arrow"
 .     ), allCells = c("PlxnD1#CTCAGAACATTACACG-1", "PlxnD1#CTTCTAATCCCTAAAG-1", 
 .     "PlxnD1#GTGGCGTCATGTATCG-1", "PlxnD1#CCGCATTAGTCCAGAG-1", 
 .     "PlxnD1#CTAGGATCAGGCTACC-1", "PlxnD1#ACCCAAATCCGCGATG-1", 
 .     "PlxnD1#TAGCTTTCAGAGAGAA-1", "PlxnD1#TCAAGGTCAGCGCGTT-1", 
 .     "PlxnD1#TTCGCGTGTCGATTAC-1", "PlxnD1#AAAGATGAGATGGCAC-1", 
 .     "PlxnD1#GCAGATTGTCATTGGT-1", "PlxnD1#AGCTATGTCTAAACGC-1",
